OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -62,48 +67,95 @@ public Turreted(ActorInitializer init, TurretedInfo info)
 		{
 			this.info = info;
 			TurretFacing = GetInitialTurretFacing(init, info.InitialFacing);
+
+			if (info.RandomRealign)
+				randomFacing = init.Self.World.SharedRandom.Next(0, 256);
```

Comment:
The action to turn the turret to a random angle is pretty much an equivalent to an animation in this case to me. The point was in the _when-to-do-this-action_ aspect and not in the _what-is-this_ aspect, iow that you could just implement `INotifyIdle` and `INotifyBecomingIdle` adding a timer, and move all the `randomFacing` changes to `TickIdle` after the timer expired.